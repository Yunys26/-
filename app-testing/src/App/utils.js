// Преобразвывает строку в массив и изменяет тип строки на цифры
const addNewArrNum = (str) => {
    const newArrayString = str.split(', ');

    const newArrayNum = newArrayString.map((el) => Number(el));

    return newArrayNum;
}

// Возвращает массив
const numSeven = (arr) => {
    const newArr = arr.filter((el) => el % 7 === 0);
    console.log(newArr)
    return newArr;
}
// Возвращает boolean
const arithmeticProgression = (arr) => {

    const step = arr[1] - arr[0];
    let valid = false;

    arr.map((el, index) => {
        if (index !== arr.length - 1) {
            valid = el + step === arr[index + 1]
        }
    });

    return valid;
}


export {
    numSeven,
    addNewArrNum,
    arithmeticProgression,
}