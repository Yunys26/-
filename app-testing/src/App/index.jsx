import React from 'react';
import { numSeven, addNewArrNum, arithmeticProgression,} from './utils'
import './style.css';

const App = () => {

    const [input, setInput] = React.useState('');
    const [messageUser, setMessageUser] = React.useState('');
    const [errorBorder, setErrorBorder] = React.useState(null);

    const regex = /^((\d\d?\d?\d?\d?)(,\s)?)+$/;

    let validateInput = input.match(regex);

    const handleChangeInput = React.useCallback((event) => setInput(event.target.value), []);
    
    const handleSubmit = React.useCallback((event) => {
        event.preventDefault();

        if (input === '') {
            setMessageUser('Вы ничего не ввели');
            setErrorBorder({border : '2px solid red'});

        } else if (addNewArrNum(input).length < 10) {
            setMessageUser(`Нужно ввести ${10 - addNewArrNum(input).length } элементов`);
            setErrorBorder({border : '2px solid red'});

        } else if (addNewArrNum(input).length > 10) {
            setMessageUser(`Нужно удалить ${addNewArrNum(input).length - 10} элемента`);
            setErrorBorder({border : '2px solid red'});

        } else if (validateInput === null) {
            setMessageUser('Комбинации не соотвествует условию');
            setErrorBorder({border : '2px solid red'});
        } else if(!arithmeticProgression(numSeven(addNewArrNum(input)))) {
            setMessageUser('Комбинации не соотвествует условию');
            setErrorBorder({border : '2px solid red'});
        } else if (arithmeticProgression(numSeven(addNewArrNum(input)))) {
            setMessageUser('');
            setErrorBorder({border : '2px solid green'});
        }

    }, [input, validateInput]);

    return (
        <div className="container">
            <header>
                <div className="title">
                    <h1>Введите элементы массива</h1>
                </div>
            </header>
            <main>
                <form onSubmit={handleSubmit}>
                    <input
                        style={errorBorder}
                        type="text"
                        className="main__input"
                        placeholder="Введите элементы"
                        onChange={handleChangeInput}
                        value={input}
                    />
                    <button
                        type="submit"
                        className="main__button btn btn1"
                    >
                        Получить результат
                    </button>
                </form>
            </main>
            <footer>
                <div className="footer-text__error">
                    <p>{messageUser}</p>
                </div>
            </footer>
        </div>
    );
};

export default App;
