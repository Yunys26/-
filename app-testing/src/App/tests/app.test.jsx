import React from 'react'
import App from '..'
import { shallow } from 'enzyme'

describe('Тестирование компоненты App', () => {

    const component = shallow(<App />);

    it('Проверка на рендер', () => {
        expect(component.length).toBe(1);
    });

    it('Присутсвует ли claasName container', () => {
        expect(component.find('.container')).toHaveLength(1);
    });

    it('Присутсвует ли тэг header', () => {
        expect(component.find('header')).toHaveLength(1);
    });
    
    it('Присутсвует ли claasName title', () => {
        expect(component.find('.title')).toHaveLength(1);
    });

    it('Присутсвует ли заголовок с текстом', () => {
        expect(component.find('h1').at(0).text()).toEqual('Введите элементы массива');
    });

    it('Присутсвует ли тэг main', () => {
        expect(component.find('main')).toHaveLength(1);
    });

    it('Присутсвует ли тэг form', () => {
        expect(component.find('form')).toHaveLength(1);
    });

    it('Присутсвует ли тэг input', () => {
        expect(component.find('input')).toHaveLength(1);
    });

    it('Присутсвует ли claasName main__input', () => {
        expect(component.find('.main__input')).toHaveLength(1);
    });

    it('Присутсвует ли тэг button', () => {
        expect(component.find('button')).toHaveLength(1);
    });

    it('Присутсвует ли claasName main__button', () => {
        expect(component.find('.main__button')).toHaveLength(1);
    });

    it('Присутсвует ли кнопка с текстом', () => {
        expect(component.find('button').at(0).text()).toEqual('Получить результат');
    });

    it('Присутсвует ли тэг footer', () => {
        expect(component.find('footer')).toHaveLength(1);
    });

    it('Присутсвует ли claasName footer-text__error', () => {
        expect(component.find('.footer-text__error')).toHaveLength(1);
    });

    it('SnapShots component App', () => {
        expect(component).toMatchSnapshot();
    });
});
