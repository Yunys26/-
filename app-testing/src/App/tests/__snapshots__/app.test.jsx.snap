// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компоненты App SnapShots component App 1`] = `
<div
  className="container"
>
  <header>
    <div
      className="title"
    >
      <h1>
        Введите элементы массива
      </h1>
    </div>
  </header>
  <main>
    <form
      onSubmit={[Function]}
    >
      <input
        className="main__input"
        onChange={[Function]}
        placeholder="Введите элементы"
        style={null}
        type="text"
        value=""
      />
      <button
        className="main__button btn btn1"
        type="submit"
      >
        Получить результат
      </button>
    </form>
  </main>
  <footer>
    <div
      className="footer-text__error"
    >
      <p />
    </div>
  </footer>
</div>
`;
